src/components/DiagramComponent.vue

```diff
@@ -70,6 +70,10 @@ export default {
         return;
       }
 
+      if (codeObject.type === CodeObjectType.FUNCTION) {
+        codeObject = codeObject.classObject;
+      }
```

Comment:
The `else` block from the following `if` statement is supposed to handle this case (https://github.com/applandinc/appmap-js/blob/273d71ae819d14c27e4200a1f73778dafe4839df/src/components/DiagramComponent.vue#L80-L82). Adding this statement here short circuits that logic, but do you know why it wasn't working previously?

Reply:
I spent some time to debug why it didn't work and this is what I discovered:
when we expand first parent:
https://github.com/applandinc/appmap-js/blob/923a4021fe137c4107042f07a897b6d03295b2aa/src/components/DiagramComponent.vue#L87

function `highlightSelectedComponent` is called again:
https://github.com/applandinc/appmap-js/blob/923a4021fe137c4107042f07a897b6d03295b2aa/src/components/DiagramComponent.vue#L110

but handler is called too with event (which you selected in Trace view):
https://github.com/applandinc/appmap-js/blob/923a4021fe137c4107042f07a897b6d03295b2aa/src/components/DiagramComponent.vue#L49-L53

so `highlightSelectedComponent` is called once again with function code object, since we doesn't show functions in Dependency Map it's not highlighted